lib/ansible/modules/clustering/k8s/k8s_auth.py

```diff
@@ -0,0 +1,335 @@
+    - If set to I(absent) attempt to log out by revoking the authentication token specified in C(api_key).
+    default: present
+    choices:
+    - present
+    - absent
+  host:
+    description:
+    - Provide a URL for accessing the API server.
+    required: true
+  username:
+    description:
+    - Provide a username for authenticating with the API server.
+  password:
+    description:
+    - Provide a password for authenticating with the API server.
+  ssl_ca_cert:
+    description:
+    - "Path to a CA certificate file used to verify connection to the API server. The full certificate chain
+      must be provided to avoid certificate validation errors."
+  verify_ssl:
+    description:
+    - "Whether or not to verify the API server's SSL certificates."
+    type: bool
+  api_key:
+    description:
+    - When C(state) is set to I(absent), this specifies the token to revoke.
+
+requirements:
+  - python >= 2.7
+  - urllib3
+  - requests
+  - requests-oauthlib
+'''
+
+EXAMPLES = '''
+- block:
+  # It's good practice to store login credentials in a secure vault and not
+  # directly in playbooks.
+  - include_vars: k8s_passwords.yml
+
+  - name: Log in (obtain access token)
+    k8s_auth:
+      host: https://k8s.example.com/
+      ssl_ca_cert: ca.pem
+      username: admin
+      password: "{{ k8s_admin_password }}"
+    register: k8s_auth_results
+
+  - name: Preserve auth info as both a fact and a yaml anchor for easy access later
+    # Both the fact and the anchor are called 'k8s_auth_params'
+    set_fact:
+      k8s_auth_params: &k8s_auth_params
+        host: "{{ k8s_auth_results.k8s_auth.host }}"
+        ssl_ca_cert: "{{ k8s_auth_results.k8s_auth.ssl_ca_cert }}"
+        verify_ssl: "{{ k8s_auth_results.k8s_auth.verify_ssl }}"
+        api_key: "{{ k8s_auth_results.k8s_auth.api_key }}"
+
+  # Previous task generated I(k8s_auth) fact, which you can then use
+  # in k8s modules like this:
+  - name: Get a list of all pods from any namespace
+    k8s_facts:
+      <<: *k8s_auth_params
+      kind: Pod
+    register: pod_list
+
+  always:
+  - name: If login succeeded, try to log out (revoke access token)
+    when: k8s_auth_params is defined
+    k8s_auth:
+      state: absent
+      <<: *k8s_auth_params
+'''
+
+RETURN = '''
+k8s_auth:
+  description: Kubernetes authentication facts.
+  returned: success
+  type: complex
+  contains:
+    api_key:
+      description: Authentication token.
+      returned: success
+      type: str
+    host:
+      description: URL for accessing the API server.
+      returned: success
+      type: str
+    ssl_ca_cert:
+      description: Path to a CA certificate file used to verify connection to the API server.
+      returned: success
+      type: str
+    verify_ssl:
+      description: "Whether or not to verify the API server's SSL certificates."
+      returned: success
+      type: bool
```

Comment:
Needs a `default: true` to match the argument spec.

Reply:
I've changed it, but it looks really weird to me…

Reply:
Ok, please disregard that comment, fixed now.